### ISSUE TITLE
[Feature Request] Option to Automatically Compress Image/Video After Upload

### ISSUE THREAD

**Issue by 8227846265** [ente-io/photos-app#1688]
**Description:**
Please consider adding a feature in the settings that allows users to upload images and videos, which are then compressed to save space. 

- Ideally, this feature would compress the files after uploading without any loss in image or video quality. That would be even better.
- However, if compression after uploading is not possible due to end-to-end encryption (E2EE), then compressing before uploading could be an alternative approach.

**Applicable to:**
- Desktop App
- Mobile App
- Web App

**Open Source Technology Reference**
- https://github.com/caydey/ffshare
- https://github.com/short-pixel-optimizer/

**Comment by vishnukvmd** [comment]
Given how the resolution of displays is improving year after year, we're not fans of lossy compression. It's important to have the ability to rediscovering memories in their original quality and preserving every pixel reduces the scope for regrets on that front.

That said, we understand the use case for saving space. I'll keep this issue open so that community can chime in with their feedback.

**Comment by windmueller** [comment]
I would like to see this feature, too. Smartphones just do not have the computing power for an efficient on-the-fly compression of videos. My Pixel 7 produces H.264 output in 1920x1080 with roughly 250 MB/s. Compressing this offline with

```
ffmpeg -i in.mp4 -vcodec libx265 out.mp4
```

reduces the size to 1/10th (25 MB/s) with hardly any noticeable loss in quality.

**Comment by vishnukvmd** [comment]
@windmueller could you please check out [ffshare](https://github.com/caydey/ffshare/) on your Pixel 7 and let us know if the performance is acceptable?

**Comment by windmueller** [comment]
Of course. I gave the version available in F-Droid a try. It only supports H.264 and will produce a video with about 50 MB/s using the default settings. Unfortunately I was not able to finish my test because the app [will stop working](https://github.com/caydey/ffshare/issues/4) if it is not constantly in the foreground.

However, I would suspect that using the same codec would result in less quality reductions, so even with the doubled size I would think of this as a good compromise.

**Comment by windmueller** [comment]
Hm, I just realized that Videos encoded with H.265 can only be opened on my smartphone. Neither the desktop app nor the web gallery in Firefox are able to play the file.
